src/components/MetadataSources/SourceManagement/Contact/contact-util.js

```diff
@@ -20,9 +18,12 @@ export const handleContactSelected = (fields, index, selectedContact) => {
     cName = get(selectedContact.personal, 'lastName', '') + ', ' + get(selectedContact.personal, 'firstName', '');
   }
 
-  onUpdateField(fields, index, {
-    externalId: cId,
-    name: cName,
-    type: cPlugin,
+  fields.update(index, {
+    ...fields.value[index],
```

Comment:
You might want to include a test for this spread operator (test that existing attributes are kept when updating).
This can be tested by adding an additional attribute to the `value` array:
```
{ externalId: 'old-id', name: 'old-name', type: 'old-type', role: 'librarian' }
```
```
expect(doUpdate).toHaveBeenCalledWith(0, {
      externalId: 'user-1',
      name: 'Smith, Anna',
      type: 'contact',
      role: 'librarian'
    });
```